### PR TITLE
[RPD-248] Add a basic example demonstrating how to use the API

### PR DESCRIPTION
**Pull request by swells2020**
This PR updates the Matcha documentation to demonstrate an alternative API implementation for the `matcha-examples` repo.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Other (add details above)

### CODE REVIEW

**Review by Christopher-Norman — APPROVED** [review]
LGTM

File: docs/getting-started.md [hunk 1]
```diff
@@ -78,16 +80,27 @@ When you run this command, you'll be taken to the Azure login screen in a web br
...
 ```
 
-Initially, Matcha will ask you a few questions about how you'd like your infrastructure to be set up. Specifically, it will ask for a _name_ for your infrastructure, a _region_ to deploy it to. Once these details are provided, Matcha will proceed to initialize a remote state manager and ask for a password. After that, it will go ahead of provision infrastructure.
+> Note: users have the choice of passing optional arguments representing the location, prefix, and password parameters by using '--location', '--prefix', or '--password'. For example; `--location uksouth --prefix test123 --password strong_password`. 
+
+API:
+```python
+import matcha_ml.core as matcha
+
+_ = matcha.provision(location="uksouth", prefix="test123", password="strong_password")
```
Comment (JonoCX) [inline docs/getting-started.md:94]: Might be worth specifying the variable that's returned for clarity: `matcha_state: MatchaState = ...`

File: docs/getting-started.md [hunk 2]
```diff
@@ -152,6 +165,23 @@ Experiment tracker
 
 By default, Matcha will hide sensitive resource properties. If you need one of these properties, then you can add the `--show-sensitive` flag to your `get` command.
 
+API:
+```python
+import matcha_ml.core as matcha
+
+_ = matcha.get()
```
Comment (JonoCX) [inline docs/getting-started.md:172]: Similar here: `matcha_state: MatchaState = ...`

File: docs/getting-started.md [hunk 3]
```diff
@@ -152,6 +165,23 @@ Experiment tracker
...
+```
+
+As with the CLI tool, users have the ability to 'get' specific resources by passing optional `resource_name` and `property_name` arguments to the get function, as demonstrated below:
+
+```python
+import matcha_ml.core as matcha
+
+_ = matcha.get(resource_name="experiment_tracker", property_name="flavor")
```
Comment (JonoCX) [inline docs/getting-started.md:180]: Here too

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice one - see comments, which I think will improve clarity for readers

File: docs/getting-started.md (hunk 1 above)
Comment (swells2020) [reply docs/getting-started.md]: Good idea: added.

**Review by JonoCX — APPROVED** [review]

**Pull request merged**